src/palace.pyx

```diff
@@ -119,6 +120,27 @@ def query_extension(name: str) -> bool:
     return devmgr.query_extension(name)
 
 
+def current_context() -> Optional[Context]:
+    """Return the context that is currently used.
+
+    Note that the message handler of the context haven't
+    been copied properly. That means::
+
+        with context:
+            assert current_context().message_handler \
+                   is context.message_handler
+            # failed
```

Comment:
This should be in an new issue.

Reply:
Let's do another PR for that shall we?

Reply:
Yep if you're interested, but if you want to document it in the docstring keep it in human language.  Still need an issue for it tho.